From 8d7cac0bf9f9775cae1d43de7138206758f28b0c Mon Sep 17 00:00:00 2001
From: jeanlf <jeanlf@gpac.io>
Date: Thu, 25 Apr 2024 16:10:14 +0200
Subject: [PATCH] Fixed #2810

---
 src/filters/ff_mx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/filters/ff_mx.c b/src/filters/ff_mx.c
index 6c704dce25..45499b4f5c 100644
--- a/src/filters/ff_mx.c
+++ b/src/filters/ff_mx.c
@@ -247,7 +247,7 @@ static GF_Err ffmx_init_mux(GF_Filter *filter, GF_FFMuxCtx *ctx)
 }
 
 
-#if (LIBAVFORMAT_VERSION_MAJOR <= 59)
+#if (LIBAVFORMAT_VERSION_MAJOR < 61)
 static int ffavio_write_packet(void *opaque, uint8_t *buf, int buf_size)
 #else
 static int ffavio_write_packet(void *opaque, const uint8_t *buf, int buf_size)
