diff -uNrp a/src/filters/ff_common.h b/src/filters/ff_common.h
--- a/src/filters/ff_common.h	2022-12-19 16:03:06.000000000 +0000
+++ b/src/filters/ff_common.h	2023-02-28 15:41:54.481172893 +0000
@@ -46,7 +46,7 @@
 #define FF_FREE_PCK(_pkt)	av_free_packet(_pkt);
 #define FF_RELEASE_PCK(_pkt)
 #define FF_INIT_PCK(ctx, _pkt) { pkt = &ctx->pkt; av_init_packet(pkt); }
-#define FF_OFMT_CAST	(AVOutputFormat *)
+#define FF_OFMT_CAST	(FFOutputFormat *)
 #define FF_IFMT_CAST	(AVInputFormat *)
 #else
 #define FF_FREE_PCK(_pkt)	av_packet_unref(_pkt);
diff -uNrp a/src/filters/ff_mx.c b/src/filters/ff_mx.c
--- a/src/filters/ff_mx.c	2022-12-19 16:03:06.000000000 +0000
+++ b/src/filters/ff_mx.c	2023-02-28 15:42:44.758941007 +0000
@@ -229,7 +229,7 @@ static GF_Err ffmx_open_url(GF_FFMuxCtx
 {
 	const char *dst;
 	Bool use_gfio=GF_FALSE;
-	const AVOutputFormat *ofmt = ctx->muxer->oformat;
+	const FFOutputFormat *ofmt = ctx->muxer->oformat;
 
 	dst = final_name ? final_name : ctx->dst;
 	if (!strncmp(dst, "gfio://", 7)) {
@@ -280,7 +280,7 @@ static GF_Err ffmx_open_url(GF_FFMuxCtx
 static GF_Err ffmx_initialize_ex(GF_Filter *filter, Bool use_templates)
 {
 	const char *url, *sep;
-	const AVOutputFormat *ofmt;
+	const FFOutputFormat *ofmt;
 	GF_FFMuxCtx *ctx = (GF_FFMuxCtx *) gf_filter_get_udta(filter);
 
 	if (!ctx->dst) return GF_BAD_PARAM;
@@ -1358,7 +1358,7 @@ static GF_FilterProbeScore ffmx_probe_ur
 		return GF_FPROBE_NOT_SUPPORTED;
 
 
-	const AVOutputFormat *ofmt = av_guess_format(NULL, url, mime);
+	const FFOutputFormat *ofmt = av_guess_format(NULL, url, mime);
 	if (!ofmt && mime) ofmt = av_guess_format(NULL, NULL, mime);
 	if (!ofmt && url) ofmt = av_guess_format(NULL, url, NULL);
 
