diff -up gpac/configure.libdir gpac/configure
--- gpac/configure.libdir	2012-06-23 17:35:19.000000000 +0200
+++ gpac/configure	2012-06-23 19:12:16.815152617 +0200
@@ -2932,7 +2932,7 @@ echo '	$(CC) $(CFLAGS) $(PIC_CFLAGS) -c
 #pkg-config
 echo "prefix=$prefix"				 > gpac.pc
 echo "exec_prefix=\${prefix}"			>> gpac.pc
-echo "libdir=\${exec_prefix}/lib"		>> gpac.pc
+echo "libdir=\${exec_prefix}/$libdir"		>> gpac.pc
 echo "includedir=\${exec_prefix}/include"	>> gpac.pc
 echo ""						>> gpac.pc
 echo "Name: gpac"				>> gpac.pc
