diff -up gpac/applications/generators/SVG/Makefile.libxml2 gpac/applications/generators/SVG/Makefile
--- gpac/applications/generators/SVG/Makefile.libxml2	2008-12-17 12:59:23.000000000 +0100
+++ gpac/applications/generators/SVG/Makefile	2008-12-17 12:59:23.000000000 +0100
@@ -37,7 +37,7 @@ SRCS := $(OBJS:.o=.c) 
 all: $(PROG)
 
 SVGGen$(EXE): $(OBJS) 
-	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS) $(XML2_LFLAGS) $(EXTRALIBS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS) $(XML2_LIBS) $(EXTRALIBS)
 
 
 %.o: %.c
diff -up gpac/configure.libxml2 gpac/configure
--- gpac/configure.libxml2	2008-12-17 12:59:23.000000000 +0100
+++ gpac/configure	2008-12-17 13:02:37.000000000 +0100
@@ -68,6 +68,7 @@ has_alsa="no"
 has_jack="no"
 has_x11="no"
 has_x11_shm="no"
+has_libxml2="no"
 has_x11_xv="no"
 disable_svg="no"
 no_gcc_opt="no"
@@ -901,6 +902,15 @@ fi
 fi
 #end wx test
 
+# If svg isn't disabled
+if test "$disable_svg" != "yes"; then
+# Then we check libxm2 presence via pkg-config
+  if pkg-config libxml-2.0 --exists > /dev/null 2>&1  ; then
+    libxml2_cflags=`pkg-config libxml-2.0 --cflags`
+    libxml2_lib_flags=`pkg-config libxml-2.0 --libs`
+    has_libxml2="yes"
+  fi
+fi
 
 # look for IPv6
 cat > $TMPC << EOF
@@ -1579,6 +1589,11 @@ echo "CONFIG_JACK=$has_jack" >> config.m
 echo "CONFIG_A52=$has_a52" >> config.mak
 
 echo "DISABLE_SVG=$disable_svg" >> config.mak
+echo "HAS_LIBXML2=$has_libxml2" >> config.mak
+if test "$has_libxml2" = "yes"; then
+  echo "XML2_CFLAGS=$libxml2_cflags" >> config.mak
+  echo "XML2_LIBS=$libxml2_lib_flags" >> config.mak
+fi
 echo "GPAC_USE_TINYGL=$has_tinygl" >> config.mak
 echo "OGL_INCLS=$INCL3D" >> config.mak
 if test "$disable_svg" = "yes" ; then
