diff -up gpac-2.2.1/src/utils/gzio.c.orig gpac-2.2.1/src/utils/gzio.c
--- gpac-2.2.1/src/utils/gzio.c.orig	2023-04-24 14:57:53.000000000 +0200
+++ gpac-2.2.1/src/utils/gzio.c	2024-03-12 11:40:34.486536168 +0100
@@ -407,7 +407,7 @@ int gf_gzread(void *file, voidp buf, uns
 			uInt n = s->stream.avail_in;
 			if (n > s->stream.avail_out) n = s->stream.avail_out;
 			if (n > 0) {
-				zmemcpy(s->stream.next_out, s->stream.next_in, n);
+				memcpy(s->stream.next_out, s->stream.next_in, n);
 				next_out += n;
 				s->stream.next_out = next_out;
 				s->stream.next_in   += n;
