diff -up gpac-2.2.1/src/filters/dec_nvdec_sdk.c.orig gpac-2.2.1/src/filters/dec_nvdec_sdk.c
--- gpac-2.2.1/src/filters/dec_nvdec_sdk.c.orig	2023-04-24 12:57:53.000000000 +0000
+++ gpac-2.2.1/src/filters/dec_nvdec_sdk.c	2024-03-12 11:42:51.690883435 +0000
@@ -197,7 +197,7 @@ tcuvidDecodePicture                   *c
 tcuvidMapVideoFrame                   *cuvidMapVideoFrame;
 tcuvidUnmapVideoFrame                 *cuvidUnmapVideoFrame;
 
-#if defined(WIN64) || defined(_WIN64) || defined(__x86_64) || defined(AMD64) || defined(_M_AMD64)
+#if defined(WIN64) || defined(_WIN64) || defined(__x86_64) || defined(AMD64) || defined(_M_AMD64) || defined(__aarch64__)
 tcuvidMapVideoFrame64                 *cuvidMapVideoFrame64;
 tcuvidUnmapVideoFrame64               *cuvidUnmapVideoFrame64;
 #endif
diff -up gpac-2.2.1/src/filters/dec_nvdec_sdk.h.orig gpac-2.2.1/src/filters/dec_nvdec_sdk.h
--- gpac-2.2.1/src/filters/dec_nvdec_sdk.h.orig	2023-04-24 12:57:53.000000000 +0000
+++ gpac-2.2.1/src/filters/dec_nvdec_sdk.h	2024-03-12 11:43:53.158031385 +0000
@@ -1821,7 +1821,7 @@ typedef void *CUDADRIVER;
 
 
 
-#if defined(__x86_64) || defined(AMD64) || defined(_M_AMD64)
+#if defined(__x86_64) || defined(AMD64) || defined(_M_AMD64) || defined(__aarch64__)
 #if (CUDA_VERSION >= 3020) && (!defined(CUDA_FORCE_API_VERSION) || (CUDA_FORCE_API_VERSION >= 3020))
 #define __CUVID_DEVPTR64
 #endif
@@ -2526,7 +2526,7 @@ typedef CUresult CUDAAPI tcuvidMapVideoF
 typedef CUresult CUDAAPI tcuvidUnmapVideoFrame(CUvideodecoder hDecoder, unsigned int DevPtr);
 #endif
 
-#if defined(WIN64) || defined(_WIN64) || defined(__x86_64) || defined(AMD64) || defined(_M_AMD64)
+#if defined(WIN64) || defined(_WIN64) || defined(__x86_64) || defined(AMD64) || defined(_M_AMD64) || defined(__aarch64__)
 /**
  * \fn CUresult CUDAAPI cuvidMapVideoFrame64(CUvideodecoder hDecoder, int nPicIdx, unsigned long long *pDevPtr, unsigned int *pPitch, CUVIDPROCPARAMS *pVPP);
  * map a video frame
