diff -up gpac/include/gpac/setup.h.noinclude gpac/include/gpac/setup.h
--- gpac/include/gpac/setup.h.noinclude	2010-04-26 15:23:51.000000000 +0200
+++ gpac/include/gpac/setup.h	2010-12-12 01:42:49.088146415 +0100
@@ -33,8 +33,6 @@ extern "C" {
 This is only needed when building libgpac and modules when libgpac is not installed*/
 #ifdef GPAC_HAVE_CONFIG_H
 # include "config.h"
-#else
-# include <gpac/configuration.h>
 #endif
 
 
