Index: gpac-2.2.0/share/doc/doxyfile
===================================================================
--- gpac-2.2.0.orig/share/doc/doxyfile
+++ gpac-2.2.0/share/doc/doxyfile
@@ -765,7 +765,7 @@ WARN_FORMAT            = "$file:$line: $
 
 WARN_LOGFILE           = doxygen_warnings.txt
 
-WARN_AS_ERROR	 = FAIL_ON_WARNINGS
+WARN_AS_ERROR	 = NO
 
 
 #---------------------------------------------------------------------------
