--- Makefile.old	2008-02-11 18:27:28.000000000 -0500
+++ Makefile	2008-02-11 18:28:13.000000000 -0500
@@ -11,13 +11,13 @@
 
 V4STUDIODIR=
 INSTDIRS=mp4client
-ifeq ($(TARGET_ARCH_ARMV4L),yes)
-else
-ifeq ($(CONFIG_LINUX),yes)
-INSTDIRS+=osmozilla
-APPDIRS+=osmozilla
-endif
-endif
+#ifeq ($(TARGET_ARCH_ARMV4L),yes)
+#else
+#ifeq ($(CONFIG_LINUX),yes)
+#INSTDIRS+=osmozilla
+#APPDIRS+=osmozilla
+#endif
+#endif
 
 ifeq ($(USE_WXWIDGETS), yes)
 APPDIRS+=osmo4_wx
